src/MICore/Debugger.cs

```diff
@@ -716,6 +748,30 @@ private Task<Results> CmdUnixBreak(int debugeePid, ResultClass expectedResultCla
             return Task.FromResult<Results>(new Results(ResultClass.done));
         }
 
+        private Task<Results> CmdBreakWindows(int debugeePid, ResultClass expectedResultClass)
+        {
+            lock (this._debuggeePids)
+            {
+                foreach (int pid in this._debuggeePids.Values)
+                {
+                    try
+                    {
+                        // Mingw has no kill utility. Instead, use kernel32!DebugBreakProcess. Note that this
+                        // can deadlock if the loader lock is taken by a suspended thread. This approach does
+                        // not work in cygwin
+                        Process p = Process.GetProcessById(pid);
+                        WindowsNativeMethods.DebugBreakProcess(p.Handle);
```

Comment:
It looks like SafeHandle is available in CoreCLR


---
In reply to: [63624644](https://github.com/Microsoft/MIEngine/pull/327#discussion_r63624644) [](ancestors = 63624644)